cobalt/watchdog/watchdog.cc

```diff
@@ -769,14 +757,45 @@ void Watchdog::SetPersistentSettingWatchdogCrash(bool can_trigger_crash) {
 }
 
 bool Watchdog::LogEvent(const std::string& event) {
+  if (is_logtrace_disabled_) {
+    return true;
+  }
+
   return instrumentation_log_.LogEvent(event);
 }
 
 std::vector<std::string> Watchdog::GetLogTrace() {
+  if (is_logtrace_disabled_) {
+    return {};
+  }
+
   return instrumentation_log_.GetLogTrace();
 }
 
-void Watchdog::ClearLog() { instrumentation_log_.ClearLog(); }
+void Watchdog::ClearLog() {
+  if (is_logtrace_disabled_) {
+    return;
+  }
+
+  instrumentation_log_.ClearLog();
+}
+
+bool Watchdog::GetPersistentSettingLogtraceEnable() {
+  if (!persistent_settings_) return kDefaultSettingLogtraceEnable;
+
+  // Gets the boolean that controls whether or not LogTrace is enabled.
+  return persistent_settings_->GetPersistentSettingAsBool(
+      kPersistentSettingLogtraceEnable, kDefaultSettingLogtraceEnable);
+}
+
+void Watchdog::SetPersistentSettingLogtraceEnable(bool enable_logtrace) {
```

Comment:
Should we clear the log if someone disables log trace?

Reply:
I'd keep it simple. The setting is applied on restart anyway and by that time the log is already empty.

Reply:
Actually, if client doesn't call `getViolations()`, violations may stuck there forever. I think it's safer to clear.